src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -372,7 +372,6 @@ internal NamespaceOrTypeOrAliasSymbolWithAnnotations BindNamespaceOrTypeOrAliasS
 
                         if (ShouldCheckConstraints && constructedType.IsNullableType())
                         {
-                            ReportUseSiteDiagnostics(constructedType.TypeSymbol.OriginalDefinition, diagnostics, syntax);
```

Comment:
>ReportUseSiteDiagnostics(constructedType.TypeSymbol.OriginalDefinition, diagnostics, syntax); [](start = 28, length = 93)

It feels like this line should still be executed if SetIsAnnotated was called. We need to come up with a test scenario that breaks.

Reply:
Thanks. The use-site diagnostics were not getting reported on a `T?` local.

---
In reply to: [214494105](https://github.com/dotnet/roslyn/pull/29608#discussion_r214494105) [](ancestors = 214494105)